utils/frame/benchmarking-cli/src/storage/write.rs

```diff
@@ -62,28 +61,45 @@ impl StorageCmd {
 		let mut rng = Self::setup_rng();
 		kvs.shuffle(&mut rng);
 
+		// Generate all random values first; Make sure there are no collisions with existing
+		// db entries, so we can rollback all additions without corrupting existing entries.
+		let mut new_values = Vec::with_capacity(kvs.len());
+		for (k, original_v) in kvs.clone() {
```

Comment:
```suggestion
		for (k, original_v) in kvs.iter() {
```
or iter_mut if writing in original_v